src/components/InteractiveMap/InteractiveMap.vue

```diff
@@ -105,105 +52,125 @@
 <script lang="ts">
 import Vue from "vue";
 import axios from "axios";
-/// <reference path="@/types/escher.d.ts" />
-import * as escher from "@dd-decaf/escher";
 import * as settings from "@/settings";
+import Escher from "@/components/InteractiveMap/Escher.vue";
+import Card from "@/components/InteractiveMap/Card.vue";
 
 export default Vue.extend({
   name: "InteractiveMap",
+  components: {
+    Escher,
+    Card
+  },
   data: () => ({
     escherBuilder: null,
-    isInitializingEscher: true,
     currentMapId: null,
-    isLoadingMap: false,
-    cards: [
-      {
-        // TODO: Replace test card
-        name: "Design",
-        method: "pFBA",
-        model: { name: "iJO1366" },
-        organism: { name: "E. coli" },
-        growthRate: 0.874
-      }
-    ]
+    fluxDistribution: null,
+    mapData: null,
+    cards: []
   }),
   methods: {
+    escherLoaded() {
+      this.$store.getters["models/onData"](this.addDefaultCard);
+    },
     changeMap() {
-      this.isLoadingMap = true;
       // TODO: Get map from maps state lazy loader
       axios
         .get(`${settings.apis.maps}/maps/${this.currentMapId}`)
         .then(response => {
-          // Note that this will freeze the entire application, including
-          // progress spinners.
-          this.escherBuilder.load_map(response.data.map);
-          this.isLoadingMap = false;
+          this.mapData = response.data.map;
+        })
+        .catch(error => {
+          // TODO: show snackbar
         });
     },
-    reactionState(id: string, type?: string) {
-      // TODO
-    },
-    handleKnockout(reactionId: string) {
-      // TODO
-    },
-    handleKnockoutGenes(reactionId: string) {
-      // TODO
+    addDefaultCard() {
+      const model = this.$store.state.models.models.find(
```

Comment:
Still think this sort of thing could be easily avoided by making `$store.state.models` an object. I'd rather convert the object to an array for the table views if necessary.

Reply:
What do you refer to by "this sort of thing" that you wish to avoid? You need to handle the case of the object missing either way, either through `find` + check for `undefined` or use `key in models`. Which one doesn't matter to me, but I don't see a reason to change the datastructure to differ from what's returned from the APIs, i.e., an ordered list of objects with PK as a field in each row.

Reply:
It matters to me so I guess I'll make another PR eventually :)